src/worker/utils/chem.ts

```diff
@@ -136,6 +136,99 @@ export const getMatchingSubstructure = ({ structure, substructure }: { structure
   return { matchingAtoms: atoms, matchingBonds: bonds };
 };
 
+export const isValidMolBlock = (mdl: string) => {
+  if (!mdl.includes('M  END')) return false;
+  const mol = get_molecule(mdl, globalThis.workerRDKit);
+  if (!mol) return false;
+  try {
+    return mol.is_valid();
+  } finally {
+    release_molecule(mol);
+  }
+};
+
+/**
+ * Convert molecule structure to given notation (smiles, smarts, molblock, ...)
+ * TODO some notations like inchi don't work with qmol
+ */
+export const convertMolNotation = ({
+  moleculeString,
+  targetNotation,
+  sourceNotation,
+  useQMol = false,
+}: {
+  moleculeString: string;
+  targetNotation: MolNotation;
+  sourceNotation?: SourceMolNotation;
+  useQMol?: boolean;
+}): string | null => {
+  if (sourceNotation != null) {
+    if (sourceNotation === targetNotation) throw new Error('Source and target notations must differ');
+    if (!_validateSource(moleculeString, sourceNotation)) throw new Error('Molecule string not valid');
+  }
+  const mol = useQMol
+    ? globalThis.workerRDKit.get_qmol(moleculeString)
```

Comment:
are qmols used frequently? we might want to cache them like we are caching mols, and do a `create_qmol` `release_qmol`. But we can do this in another pr if you want

Reply:
Only when using smarts. I don't think it's necessary to use cache for qmol right now.